### DISCUSSION TITLE
Help needed. Connect FS200-SHT10 humidity sensor to Artisan using an Arduino Mega and MODBUS

### DISCUSSION THREAD

**Discussion by Robjones6146** [artisan-roaster-scope/artisan#1244] [General]
I am trying to connect a serial humidity sensor FS200-SHT10 using an Arduino MEGA 2560 (USB) via Artisan MODBUS (serial)
I want to display the humidity in the Artisan LCD on the right,

I have tried a combination of Baud rates, Function 3, 4 and other settings.
I get the error:  98 06:43:33.597 Modbus Communication Error

Any help will be greatly appreciated.

Here is the Arduino code:

```
#include <SHT1x.h>
#include <ModbusRtu.h>

// data array for modbus network sharing
uint16_t au16data[16] = { 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, -1 };

Modbus slave;

// Specify data and clock connections and instantiate SHT1x object
#define dataPin  10
#define clockPin 11

SHT1x sht1x(dataPin, clockPin);

void setup()
{
   slave =  Modbus(1,0,0); // MODBUS object declaration: (1 = slave, 0 = RS232, 0 = RS232)
   slave.begin(19200); // 19200 baud, 8-bits, none, 1-bit stop
  
   delay(500);
}

void loop()
{
  au16data[2] = ((uint16_t) sht1x.readHumidity() );
  slave.poll( au16data, 16 );

  delay(200);
}
```
Here are my Artisan settings:
![Modbus](https://github.com/artisan-roaster-scope/artisan/assets/138966151/0c0080df-d7c6-4595-9273-d62e6ceeb3f6)

![Device](https://github.com/artisan-roaster-scope/artisan/assets/138966151/eaf3a9d8-d20a-4760-aff8-c084fb0e73fa)
![SerialLog](https://github.com/artisan-roaster-scope/artisan/assets/138966151/9d6e5af9-c653-42d8-84c9-b716f637fc6f)

Thank you in advance.

![Humidity](https://github.com/artisan-roaster-scope/artisan/assets/138966151/b36827ec-6d7f-46ae-85ec-24891b832d0a)

![ArduinoConnection](https://github.com/artisan-roaster-scope/artisan/assets/138966151/7f11ac52-3e64-4d4e-9ef2-566d4061af8a)

**Comment by Robjones6146** [comment]
The FS200-SHT10 has 4 pins. Serial clock and serial data.

GND: Ground
SCK: Clock pin (input only)
DATA: Data pin (bidirectional)
VCC: Module power supply – 5V

When connected it works fine. When I try to display it with Artisan LCD I get the error. Modbus Communication Error.

**Reply by poundy** [reply]
what exactly do you mean by "when connected it works fine"? Do you mean that when you write a specific sketch to read and display that, it shows the correct values? If so, then what that means is that your use of the ModbusRTU code isn't working, or your connection to read over modbus isn't working. 
Have you looked at an example of how to use ModbusRTU and compared that? https://github.com/norgor/ModbusRTU/blob/master/examples/ModbusRTU_Example/ModbusRTU_Example.ino for example, appears to me (after a very quick comparison to your code) to do many additional things; I'm no arduino expert but I don't know if this is the now-included modbus library you're referencing, and if so you can see a different way to initialise the slave.

**Reply by Robjones6146** [reply]
what exactly do you mean by "when connected it works fine"? The output from the Arduino works fine. It works, no problem with the Arduino sketch as far as talking to the probe and reporting the data.

Yes, my issue is a Modbus Communication Error. I will look at your suggestion, I don't really follow what's going on in the code, but thank you!

I have read and tried many resources related to Arduino and Modbus. None deal with what I want specifically, but some are close.
https://lukasgrasse.medium.com/how-to-make-an-arduino-controlled-coffee-roaster-f6a3334fd7d5
https://artisan-roasterscope.blogspot.com/2013/03/modbus-rtu.html
https://artisan-roasterscope.blogspot.com/2013/05/more-modbus.html
https://www.home-barista.com/roasting/getting-artisan-to-talk-to-arduino-t58234.html
.... and a few others.

**Reply by Robjones6146** [reply]
I have tried the example. I don't know what it does or what it's for. It's also a different library that I am using.

It gives the same error anyway.

The library I am using for the sensor.
https://github.com/practicalarduino/SHT1x/blob/master/SHT1x.h
The library I am using for MODBUS
https://github.com/smarmengol/Modbus-Master-Slave-for-Arduino/blob/master/ModbusRtu.h

**Comment by Robjones6146** [comment]
I have also tried Scan  - function 4 as I want to use read-only. No results.
![scan](https://github.com/artisan-roaster-scope/artisan/assets/138966151/51d81f49-92e0-48ea-8d19-d14f81f20faf)
